consensus/context/src/difficulty.rs

```diff
@@ -328,8 +328,7 @@ fn next_difficulty(
         time_span = 1;
     }
 
-    // TODO: do checked operations here and unwrap so we don't silently overflow?
-    (windowed_work * u128::from(hf.block_time().as_secs()) + time_span - 1) / time_span
+    (windowed_work * u128::from(hf.block_time().as_secs())).div_ceil(time_span)
 }
```

Comment:
```suggestion
    windowed_work
        .checked_mul(u128::from(hf.block_time().as_secs()))
        .unwrap()
        .div_ceil(time_span)
}
```
```suggestion
    // TODO: do `checked_mul` here and unwrap so we don't silently overflow?
    (windowed_work * u128::from(hf.block_time().as_secs())).div_ceil(time_span)
}
```

Reply:
I'm still sure if unwrapping is the best option, so I am going to keep the TODO